src/editors/editor.ts

```diff
@@ -1,6 +1,24 @@
-declare interface Editor {
+export default abstract class Editor implements IEditor {
+  abstract name: string;
+
+  abstract icon: string;
+
+  get key(): string {
+    return this.name.replace(/\s/g, "").toLowerCase();
+  }
+
+  abstract isEditorInstalled(): Promise<boolean>;
+
+  abstract isPluginInstalled(): Promise<boolean>;
+
+  abstract installPlugin(): Promise<void>;
+
+  abstract uninstallPlugin(): Promise<void>;
+}
+
+declare interface IEditor {
```

Comment:
How about `EditorInterface`?

Reply:
It's most common to prefix with `I` for interfaces. Do you still prefer to add the suffix `Interface`?

Reply:
That's a C# convention, but TypeScript follows JavaScript conventions.

Reply:
done!